### ISSUE TITLE
k-fold grouped

### ISSUE THREAD

**Issue by jayrobwilliams** [paul-buerkner/brms#619]
Currently `kfold` with `folds = "stratified"` places observations in the same `group` in different `folds`. The development version of `loo` now includes a function `loo::kfold_split_grouped` that will construct folds such that entire groups are either included or excluded in the folds. I've modified `kfold_internal` to allow *k*-fold cross-validation that leaves entire groups out using `loo::kfold_split_grouped`.

I've also modified `kfold_internal` to return groupwise instead of pointwise elpd standard errors following @avehtari's demo for `rstanarm` [here](https://avehtari.github.io/modelselection/rats_kcv.html#53_grouped_k-fold_for_leave-one-group-out). 

I've done some limited tests and so far it doesn't look like anything I've done has broken `kfold_predict`. The [modified function](https://github.com/jayrobwilliams/brms/blob/kfold-grouped/R/kfold-helpers.R) can be viewed on [my forked repo](https://github.com/jayrobwilliams/brms) under the `kfold-grouped` branch. Would you have any interest in incorporating this functionality into `brms`?

**Comment by paul-buerkner** [comment]
Thanks! This is definitely an important addition to brms and I will surely implement it. Currently, the up-to-date version of `kfold_internal` is on the `loo-compare` branch, which I will merge into master once the new loo version is on CRAN (hopefully tomorrow or so). Once that is done, I will add `"grouped"` as an option in argument `folds`.

**Comment by jayrobwilliams** [comment]
Awesome!

**Comment by paul-buerkner** [comment]
This feature is now available on the `loo-compare` branch, which is to be merged soon.
